### PR TITLE
Add smoke tests

### PR DESCRIPTION
**Pull request by ErikSchierboom**
It turned out that the Elm representer was broken. This PR fixes that and in the process, also adds a smoke test that will do an e2e test using the Docker image.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -36,32 +36,27 @@ jobs:
...
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825
         with:
           context: .
-          file: ./Dockerfile
           push: false
-          tags: |
-            ${{ github.event.repository.full_name }}:latest
-            ${{ github.event.repository.full_name }}:${{ github.sha }}
-          cache-from: type=local,src=/tmp/.buildx-cache
-          cache-to: type=local,dest=/tmp/.buildx-cache
```
Comment (ErikSchierboom) [inline .github/workflows/tests.yml:67]: We no longer need this, as we can use `type-gha` for caching

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,6 @@ src/main.js
 .DS_Store
 .coverage
 /example-output/
+/smoke-tests/**/*/mapping.json
+/smoke-tests/**/*/representation.txt
+/smoke-tests/**/*/Representation.elm
```
Comment (ErikSchierboom) [inline .gitignore:20]: This is an intermediate file, but it is safer to just add it

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,12 @@
 FROM node:lts-alpine AS builder
 
 # Working directory as specified by exercism
-WORKDIR /opt/elm-representer
+WORKDIR /opt/representer
```
Comment (ErikSchierboom) [inline Dockerfile:4]: This is the default directory almost all representers use

File: bin/run-in-docker.sh [hunk 4]
```diff
@@ -34,10 +34,10 @@ SLUG="$1"
 INPUT_DIR="$2"
 OUTPUT_DIR="$3"
 
-docker build -t elm-representer .
+docker build -t exercism/elm-representer .
```
Comment (ErikSchierboom) [inline bin/run-in-docker.sh:37]: This will re-use the cached image when `run-tests-in-docker.sh` has been run

File: bin/run.sh [hunk 5]
```diff
@@ -9,26 +9,29 @@ INPUT_DIR="$2"
 OUTPUT_DIR="$3"
 
 # Normalize identifiers and remove comments
-ELM_FILES=$(ls "$INPUT_DIR"/*.elm)
```
Comment (ErikSchierboom) [inline bin/run.sh:12]: The main problem with the representer was that Elm files are contained in the exercise's `src` directory, not in its root

**Review by ErikSchierboom — COMMENTED** [review]
Feel free to add more smoke tests and/or rename the existing smoke test directory

File: bin/run.sh (hunk 5 above)
Comment (mpizenberg) [reply bin/run.sh:12]: Thanks for catching this and for the rest of the fixes!

**Review by mpizenberg — APPROVED** [review]

**Comment by ceddlyburge** [comment]
Yep thanks @ErikSchierboom :)

Amazing to think that this has never worked and we didn't notice!

**Pull request merged**